File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/question/UpdateQuestionRequest.java
package org.example.olympic_ot_project.dto.exam.question;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import org.example.olympic_ot_project.core.QuestionLevel;
import org.example.olympic_ot_project.core.QuestionType;
import org.example.olympic_ot_project.dto.exam.option.CreateQuestionOptionRequest;

import java.util.List;

@Data
public class UpdateQuestionRequest {
    @NotBlank(message = "Nội dung không được để trống")
    private String content;

    @NotNull(message = "Loại câu hỏi không được để trống")
    private QuestionType type;

    @NotNull(message = "Cấp độ câu hỏi không được để trống")
    private QuestionLevel level;

    @NotNull(message = "Category không được để trống")
    private Integer categoryId;

    private String answer;

    @NotNull(message = "Điểm số câu hỏi không được để trống")
    private Integer score;

    @NotNull(message = "Thời gian trả lời không được bỏ trống")
    @Min(value = 30, message = "Thời gian phải có ít nhất 3- giây")
    private Integer timeLimit;

    private String imageUrl;
    private String videoUrl;
    private List<CreateQuestionOptionRequest> options;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/question/update_question_request.dart
import 'package:olympic_it_project/core/question_level.dart';
import 'package:olympic_it_project/core/question_type.dart';
import 'package:olympic_it_project/dto/admin_manager/question/create_question_option_request.dart';

class UpdateQuestionRequest {
  final String content;
  final QuestionType type;
  final QuestionLevel level;
  final String? answer;
  final int score;
  final int categoryId;
  final int timeLimit;
  final String? imageUrl;
  final String? videoUrl;
  final List<CreateQuestionOptionRequest>? options;

  UpdateQuestionRequest({
    required this.content,
    required this.type,
    required this.level,
    this.answer,
    required this.score,
    required this.categoryId,
    required this.timeLimit,
    this.imageUrl,
    this.videoUrl,
    this.options,
  });

  Map<String, dynamic> toJson() {
    final data = {
      "content": content,
      "type": type.name,
      "level": level.name,
      "answer": answer,
      "score": score,
      "categoryId": categoryId,
      "imageUrl": imageUrl,
      "videoUrl": videoUrl,
      "timeLimit": timeLimit,
    };

    if (options != null) {
      data["options"] = options!.map((option) => option.toJson()).toList();
    }

    return data;
  }
}